crates/sui-types/src/temporary_store.rs

```diff
@@ -670,7 +670,19 @@ impl<S> TemporaryStore<S> {
         }
         let cost_summary = gas_status.summary(result.is_ok());
         let gas_used = cost_summary.gas_used();
-        let gas_rebate = (cost_summary.storage_rebate as f64 * STORAGE_REBATE_RATE).round() as u64;
+        // value as computed before
+        let gas_rebate1 = (cost_summary.storage_rebate as f64 * 0.99).round() as u64;
+        // computation with u64 arithmetic
+        // multiply by STORAGE_REBATE_RATE first value (numerator)
+        let mut gas_rebate =
+            (cost_summary.storage_rebate as u128 * STORAGE_REBATE_RATE.0 as u128) as u64;
+        // divide by STORAGE_REBATE_RATE second value and round
+        gas_rebate = ((gas_rebate as u128 + (STORAGE_REBATE_RATE.1 as u128 / 2))
+            / STORAGE_REBATE_RATE.1 as u128) as u64;
+        // this is here in case we want to truncate
+        // gas_rebate = (gas_rebate as u128 / STORAGE_REBATE_RATE.1 as u128) as u64;
+        // REMOVE THIS (and related), here just to check if there are failures
+        assert_eq!(gas_rebate, gas_rebate1);
```

Comment:
this and related code will go away but it was to make sure we do not hit the assert which we did before I had rounding proper. I am not sure if you should truncate instead, pretty minor but wondering if people have opinions...